gtsfm/densify/mvs_math.py

```diff
@@ -0,0 +1,44 @@
+"""MVS math methods for gtsfm
+
+Authors: Ren Liu
+"""
+import math
+import numpy as np
+
+from gtsfm.utils.geometry_comparisons import angle_between_vectors
+
+
+def piecewise_gaussian(
+    a_x: np.ndarray, b_x: np.ndarray, theta_0: float = 5, sigma_1: float = 1, sigma_2: float = 10
+) -> float:
+    """Evaluate the similarity of measurements for the same track in camera a and camera b.
+
+    This piecewise Gaussian function outputs a float score to show the evaluation result of the similarity.
+    The total similarity between two views can be calculated by summing up the scores of all common tracks.
+
+    Details can be seen in "View Selection" paragraphs in Yao's paper https://arxiv.org/abs/1804.02505.
+
+    Args:
+        a_x: 3D coordinates of the track point in pose a, with shape (3,),
```

Comment:
which frame is `a_x` in? global or camera a's frame?

Reply:
camera a's frame

Reply:
why would we want the measurement to be at the same angle in both views? that would suggest zero parallax, which seems bad

Reply:
the parallax between two views with the highest score in this function was set to be theta_0 (default 5 degree). I think the author (Yao, MVSNet) hopes that small parallax would lead to more shared 3D points on the images from selected view pairs. These shared 3D points are constrained by both reference and source images, which will lead the cost volume to converge.

Reply:
Got it -- can you add that to the docstring? Is there a URL reference and paper section we can include?

Reply:
sure, the URL has already been included in the docstring

Reply:
Ah I see the URL now, thanks